### DISCUSSION TITLE
Why do the new WinUI3 project templates have multiple package references?

### DISCUSSION THREAD

**Discussion by mrlacey** [microsoft/WindowsAppSDK#560] [Q&A]
The [WinUI3 C# desktop project templates in the new ProjectReunion VSIX](https://marketplace.visualstudio.com/items?itemName=ProjectReunion.MicrosoftProjectReunion) contain multiple package references when only one is necessary.

The project files contain:

```xml
    <PackageReference Include="Microsoft.ProjectReunion" Version="0.5.0-prerelease" />
    <PackageReference Include="Microsoft.ProjectReunion.Foundation" Version="0.5.0-prerelease" />
    <PackageReference Include="Microsoft.ProjectReunion.WinUI" Version="0.5.0-prerelease" />
```

However, as "Microsoft.ProjectReunion" is a meta package that includes dependencies on "Microsoft.ProjectReunion.Foundation" and "Microsoft.ProjectReunion.WinUI" (as well as "Microsoft.ProjectReunion.DWrite") they are included automatically.

The projects work exactly the same way if they only have the `Microsoft.ProjectReunion` package reference.

Removing the unnecessary extra entries would not only make the project file simpler, it would also avoid the risks of inconsistencies in the future if one was updated but not the other. 
For example, if M.PR was updated to 0.8 but M.PR.WUI remained at 0.5.0 what would happen? I hope the tooling would warn about the inconsistency but it would be better if it wasn't necessary to reference both packages or make sure that both are updated at the same time.

Or am I missing something?

**Comment by dotMorten** [comment]
The same goes for the C++ templates. I've found no issues with removing them from both project types.
If they indeed are a required feature that I'm missing, then WinUI should instead make it an implicit dependency.
